cylc/uiserver/workflows_mgr.py

```diff
@@ -213,7 +214,21 @@ async def _disconnect(self, wid):
 
     async def _unregister(self, wid):
         """Unregister a workflow from the data store."""
-        self.uiserver.data_store_mgr.purge_workflow(wid)
+        store_mgr = self.uiserver.data_store_mgr
+        if wid in store_mgr.data:
+            store_mgr.update_contact(wid, status='uninstalled')
+        while True:
+            has_queues = []
+            if wid in store_mgr.delta_queues:
+                has_queues = [
+                    True
+                    for delta_queue in store_mgr.delta_queues[wid].values()
+                    if not delta_queue.empty()]
```

Comment:
I think the above can be simplified with `any()`?

Reply:
Best I could come up with, but not tested :laughing: just to show what I mean.

```py
has_queues = any(not delta_queue.empty for delta_queue in store_mgr.delta_queues[wid].values())
```

Reply:
Oh yeah, true.

Reply:
What's faster; calling `any` or list comprehension? XD

Reply:
Ok made it better, used that as the while condition.